src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas1/Cas1ExpiredApplicationsScheduledJobTest.kt

```diff
@@ -0,0 +1,81 @@
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.transaction.support.TransactionTemplate
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.events.model.ApplicationAssessedEnvelope
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.events.model.EventType
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.events.ApplicationAssessedFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.givens.`Given a Probation Region`
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ApplicationRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ApprovedPremisesApplicationEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.DomainEventType
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.ApprovedPremisesApplicationStatus
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.FeatureFlagService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.cas1.scheduled.Cas1ExpiredApplicationsScheduledJob
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.toLocalDateTime
+import java.time.Instant
+import java.time.LocalDate
+import java.util.UUID
+
+class Cas1ExpiredApplicationsScheduledJobTest : IntegrationTestBase() {
+
+  @Autowired
+  lateinit var featureFlagService: FeatureFlagService
+
+  @Autowired
+  lateinit var applicationRepository: ApplicationRepository
+
+  @Autowired
+  lateinit var transactionTemplate: TransactionTemplate
+
+  @Test
+  fun `sets the application status to EXPIRED for the correct applications when the job is run`() {
+    val user = userEntityFactory.produceAndPersist {
+      withProbationRegion(`Given a Probation Region`())
+    }
+
+    // create an application
+    val applicationId = approvedPremisesApplicationEntityFactory.produceAndPersist {
+      withCreatedByUser(user)
+      withApplicationSchema(approvedPremisesApplicationJsonSchemaEntityFactory.produceAndPersist())
+    }.id
+
+    val occurredAt = LocalDate.now().minusYears(1).minusDays(1).toLocalDateTime()
+
+    domainEventFactory.produceAndPersist {
+      withOccurredAt(occurredAt)
+      withType(DomainEventType.APPROVED_PREMISES_APPLICATION_ASSESSED)
+      withApplicationId(applicationId)
+      withData(
+        objectMapper.writeValueAsString(
+          ApplicationAssessedEnvelope(
+            id = UUID.randomUUID(),
+            timestamp = Instant.now(),
+            eventType = EventType.applicationSubmitted,
+            eventDetails = ApplicationAssessedFactory()
+              .withDecision("ACCEPTED")
+              .produce(),
+          ),
+        ),
+      )
+    }
+
+    // create a domain event for this application
+
+    Cas1ExpiredApplicationsScheduledJob(
```

Comment:
you should be able to inject this in using @Autowired as it's a spring manage bean. no need to construct yourself

Reply:
Have changed this.